storage/gcsemu/filestore.go

```diff
@@ -82,17 +115,31 @@ func (fs *filestore) GetMeta(baseUrl HttpBaseUrl, bucket string, filename string
 		return nil, fmt.Errorf("stating  %s: %w", f, err)
 	}
 
+	if (fInfo.IsDir() && !strings.HasSuffix(filename, "/")) ||
+		(!fInfo.IsDir() && strings.HasSuffix(filename, "/")) {
+		return nil, nil
+	}
+
 	return fs.ReadMeta(baseUrl, bucket, filename, fInfo)
 }
 
-func (fs *filestore) Add(bucket string, filename string, contents []byte, meta *storage.Object) error {
+func (fs *filestore) Add(bucket string, filename string, contents []byte, meta *gcsutil.Object) error {
 	f := fs.filename(bucket, filename)
-	if err := os.MkdirAll(filepath.Dir(f), 0777); err != nil {
-		return fmt.Errorf("could not create dirs for:  %s: %w", f, err)
-	}
+	if strings.HasSuffix(filename, "/") {
```

Comment:
since gcs don't have folder support hadoop file system and other file system implementations like google nio use an object with suffix "/" as a folder whil without "/" as file even so gcs treat them both as objects

Reply:
Interesting.  So would it be correct to rephrase the ask as "GCS emulator needs to support objects that end with a trailing slash `/`"?